File: eMovieFinder/eMovieFinder.Model/Dtos/Requests/Movie/MovieUpdateRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace eMovieFinder.Model.Dtos.Requests.Movie
{
    public class MovieUpdateRequest
    {
        public string? Title { get; set; }
        public DateTime? ReleaseDate { get; set; }
        public int? Duration { get; set; }
        public int? DirectorId { get; set; }
        public List<int>? Categories { get; set; }
        public List<int>? Actors { get; set; }
        public int? CountryId { get; set; }
        public string? TrailerLink { get; set; }
        public string? ImagePlainText { get; set; }
        [DefaultValue(null)]
        public byte[]? Image { get; set; }
        public string? StoryLine { get; set; }
        public double? Price { get; set; }
        public double? AverageRating { get; set; }
        public int? MovieState { get; set; }
        public decimal? Discount { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_desktop/lib/src/models/dtos/requests/movie/movie_update_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'movie_update_request.g.dart';

@JsonSerializable()
class MovieUpdateRequest {
  final String? title;
  final DateTime? releaseDate;
  final dynamic duration;
  final int? directorId;
  final List<dynamic>? categories;
  final List<dynamic>? actors;
  final int? countryId;
  final String? trailerLink;
  final String? imagePlainText;
  final String? storyLine;
  final dynamic price;
  final int? movieState;
  final dynamic discount;
  final dynamic averageRating;

  MovieUpdateRequest({
    this.title,
    this.releaseDate,
    this.duration,
    this.directorId,
    this.categories,
    this.actors,
    this.countryId,
    this.trailerLink,
    this.imagePlainText,
    this.storyLine,
    this.price,
    this.movieState,
    this.discount,
    this.averageRating
  });

  factory MovieUpdateRequest.fromJson(Map<String, dynamic> json) => _$MovieUpdateRequestFromJson(json);
  Map<String, dynamic> toJson() => _$MovieUpdateRequestToJson(this);
}